File: src/main/java/de/dk/api/codeChallenge/notification/model/dto/NotificationDto.java
package de.dk.api.codeChallenge.notification.model.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import java.util.UUID;

public record NotificationDto(
        @NotNull(message = "userId must not be null")
        UUID userId,

        @NotBlank(message = "notificationType must not be blank")
        String notificationType,

        @NotBlank(message = "message must not be blank")
        String message
) {
}

File: src/main/kotlin/de/dkb/api/codeChallenge/notification/model/dto/NotificationDto.kt
package de.dkb.api.codeChallenge.notification.model.dto


import java.util.UUID
import javax.validation.constraints.NotBlank
import javax.validation.constraints.NotNull

data class NotificationDto(
    @field:NotNull val userId: UUID,
    @field:NotBlank val notificationType: String,
    @field:NotBlank val message: String
)